packages/dd-trace/test/appsec/iast/taint-tracking/sources/sql_row.sequelize.plugin.spec.js

```diff
@@ -0,0 +1,106 @@
+                                command VARCHAR(50),
+                                createdAt DATETIME DEFAULT CURRENT_TIMESTAMP,
+                                updatedAt DATETIME DEFAULT CURRENT_TIMESTAMP )`)
+
+        await sequelize.query(`INSERT INTO examples (id, name, query, command)
+                               VALUES (1, 'Item1', 'SELECT 1', 'ls'),
+                                      (2, 'Item2', 'SELECT 1', 'ls'),
+                                      (3, 'Item3', 'SELECT 1', 'ls')`)
+      })
+
+      afterEach(() => {
+        return sequelize.close()
+      })
+
+      describe('using query method', () => {
+        testThatRequestHasVulnerability(async (req, res) => {
+          const result = await sequelize.query('SELECT * from examples')
+
+          await sequelize.query(result[0][0].query)
+
+          res.end('OK')
+        }, 'SQL_INJECTION', { occurrences: 1 }, null, null,
+        'Should have SQL_INJECTION using the first row of the result')
+
+        testThatRequestHasNoVulnerability(async (req, res) => {
+          const result = await sequelize.query('SELECT * from examples')
+
+          await sequelize.query(result[0][1].query)
+
+          res.end('OK')
+        }, 'SQL_INJECTION', null, 'Should not taint the second row of a query with default configuration')
+
+        testThatRequestHasNoVulnerability(async (req, res) => {
+          const result = await sequelize.query('SELECT * from examples')
+
+          const childProcess = require('child_process')
+          childProcess.execSync(result[0][0].command)
+
+          res.end('OK')
+        }, 'COMMAND_INJECTION', null, 'Should not detect COMMAND_INJECTION with database source')
+      })
+
+      describe('using Model', () => {
+        // let Model
+        let Example
+
+        beforeEach(() => {
+          Example = sequelize.define('example', {
+            id: {
+              type: Sequelize.DataTypes.INTEGER,
+              primaryKey: true
+            },
+            name: Sequelize.DataTypes.STRING,
+            query: Sequelize.DataTypes.STRING,
+            command: Sequelize.DataTypes.STRING
+          })
+        })
+
+        testThatRequestHasVulnerability(async (req, res) => {
+          const examples = await Example.findAll()
+
+          await sequelize.query(examples[0].query)
+
+          res.end('OK')
+        }, 'SQL_INJECTION', { occurrences: 1 }, null, null,
+        'Should have SQL_INJECTION using the first row of the result')
+
+        testThatRequestHasNoVulnerability(async (req, res) => {
+          const examples = await Example.findAll()
+
+          await sequelize.query(examples[1].query)
+
+          res.end('OK')
+        }, 'SQL_INJECTION', null, 'Should not taint the second row of a query with default configuration')
+
+        testThatRequestHasNoVulnerability(async (req, res) => {
+          const examples = await Example.findAll()
+
+          const childProcess = require('child_process')
```

Comment:
We could move this declaration to the top of the file to avoid duplication

Reply:
This is intentional, this `require` is part of the test. If I do something like this in the top of the file, not instrumented method could be loaded because the require could happen before the instrumentation.

```
const { execSync } = require('child_process')
```

This is why I usually try to do the requires into the test, when I am sure that everything is instrumented.